tests/auth_tests/test_hashers.py

```diff
@@ -460,6 +460,11 @@ def test_encode(self):
         with self.assertRaisesMessage(NotImplementedError, msg):
             self.hasher.encode('password', 'salt')
 
+    def test_decode(self):
+        msgregex = r'.*.decode\(\) method is not implemented.*'
+        with self.assertWarnsRegex(Warning, msgregex):
```

Comment:
Given that this explicitly tests the base hasher I'd replace `.*` at the beginning with the actual class name.

Reply:
My bad. I thought this would be run for all the hashers which extend `BasePasswordHasher`.